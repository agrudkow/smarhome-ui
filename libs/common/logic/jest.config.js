module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
