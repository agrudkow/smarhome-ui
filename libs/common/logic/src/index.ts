export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
