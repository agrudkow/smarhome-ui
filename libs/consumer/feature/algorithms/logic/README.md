# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
