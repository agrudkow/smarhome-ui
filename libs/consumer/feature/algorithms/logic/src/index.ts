export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
