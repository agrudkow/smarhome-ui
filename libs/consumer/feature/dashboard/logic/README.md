# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
