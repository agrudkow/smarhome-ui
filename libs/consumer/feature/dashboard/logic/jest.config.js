module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
