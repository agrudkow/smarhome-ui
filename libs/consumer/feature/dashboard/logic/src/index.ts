export * from './lib/use-billings';
