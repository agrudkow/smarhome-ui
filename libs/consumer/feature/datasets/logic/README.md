# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
