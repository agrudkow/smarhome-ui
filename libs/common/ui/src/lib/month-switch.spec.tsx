import React from 'react';
import { customRender } from '@smarthome/common/logic';
import MonthSwitch from './month-switch';

describe(' MonthSwitch', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <MonthSwitch
        month={5}
        year={2020}
        onNextMonthClick={jest.fn()}
        onPreviousMonthClick={jest.fn()}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <MonthSwitch
        month={5}
        year={2020}
        onNextMonthClick={jest.fn()}
        onPreviousMonthClick={jest.fn()}
      />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });
});
