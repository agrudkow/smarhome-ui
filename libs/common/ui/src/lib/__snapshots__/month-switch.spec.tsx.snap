// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` MonthSwitch should match snapshot 1`] = `
.c2 {
  width: 30px;
  height: 30px;
  cursor: pointer;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  color: #1BBDCE;
}

.c4 {
  width: 30px;
  height: 30px;
  cursor: pointer;
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
  color: #1BBDCE;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c3 {
  margin: 0;
  padding: 0;
  color: #444;
  font-size: 20px;
  font-weight: bold;
  line-height: 22px;
  text-align: center;
  width: 50%;
}

@media (max-width:769px) {
  .c2 {
    width: 20px;
    height: 20px;
  }
}

@media (max-width:769px) {
  .c4 {
    width: 20px;
    height: 20px;
  }
}

@media (max-width:769px) {
  .c1 {
    padding: 0;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 26px;
    line-height: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root c1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root c2"
          focusable="false"
          rotate="180deg"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <h4
      class="c3"
    >
      June
      Â 
      2020
    </h4>
    <button
      class="MuiButtonBase-root MuiIconButton-root c1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root c4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
